src/pages/ProductList/ProductList.js

```diff
@@ -1,24 +1,65 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState, useRef, useCallback } from 'react';
 import { BiUpArrow, BiDownArrow, BiFilter } from 'react-icons/bi';
+import { Link } from 'react-router-dom';
 import Dropdown from './Product/Dropdown';
 import ProductAside from './ProductAside';
 import './ProductList.scss';
 
 const ProductList = () => {
   const [dropdownShown, setDropdownShown] = useState(false);
+  // const [productMain, setProductMain] = useState([]);
 
-  const [productMain, setProductMain] = useState([]);
+  const [postList, setPostList] = useState([]); // 현재의 페이지와 생성되는 페이지를 넣기 위해서
+  const [page, setPage] = useState(1); // page를 추가하기 위해서
+  const preventRef = useRef(true); // 오류로 중복적으로 페이지 생성을 막기위한 ref
+  const obsRef = useRef(null); // 모든 글 로드 확인
+
+  const [offset, setOffset] = useState(0);
 
   const priceToString = price => {
     return price.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
   };
 
-  useEffect(() => {
-    fetch('http://10.58.52.134:3000/products/main?offset=0&limit=10')
+  const obsHandler = entries => {
+    const target = entries[0];
+    if (target.isIntersecting && preventRef.current) {
+      preventRef.current = false;
+      setPage(prev => prev + 1);
+    }
+  };
+
+  const getProductList = useCallback(() => {
+    const newPostList = [...postList];
+    if (page !== 1) {
+      setOffset(offset => offset + 9);
+    }
+    fetch(`http://10.58.52.234:3000/products/main?offset=${offset}&limit=9`)
       .then(response => response.json())
-      .then(item => setProductMain(item));
+      .then(data => {
+        if (data) {
+          setPostList(newPostList.concat(...data));
```

Comment:
concat도 좋지만
```suggestion
          setPostList([...postList, ...data]);
```
로 가도 좋을 것 같습니다!

충분히 여기서 불변성은 지켜지고 있으니 따로 newPostList를 안만드셔도 될 것 같아요!

Reply:
넵 감사합니다 해당부분 수정 완료 하였습니다.
List 페이지에서 Detail 페이지로 돌아가는 부분은 아직 합쳐지지 않아서 해당 
![스크린샷 2022-10-26 오후 8 16 48](https://user-images.githubusercontent.com/107871028/198014006-1ec8a24c-919d-4d52-838f-8cf65fec2c00.png)
![스크린샷 2022-10-26 오후 8 31 01](https://user-images.githubusercontent.com/107871028/198015444-1db650ef-3440-4ec7-8f28-12757593b510.png)